source/util/timer.h

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2018 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Contains utils for getting resource utilization
+
+#ifndef LIBSPIRV_UTIL_TIMER_H_
+#define LIBSPIRV_UTIL_TIMER_H_
+
+#if defined(SPIRV_ANDROID) || defined(SPIRV_LINUX) || defined(SPIRV_MAC) || \
+    defined(SPIRV_FREEBSD)
+
+#define SPIRV_TIMER_ENABLED
+#define SPIRV_TIMER_DESCRIPTION(out) spvutils::TimerPrintDescription(out)
+#define SPIRV_TIMER_SCROPED(out, tag) \
+  spvutils::ScropedTimer timer##__LINE__(out, tag)
+
+#if defined(SPIRV_LINUX)
+#define SPIRV_MEMORY_MEASUREMENT_ENABLED
+#endif  // defined(SPIRV_LINUX)
+
+#include <sys/resource.h>
+
+namespace spvutils {
+
+void TimerPrintDescription(std::ostream* out);
+
+class Timer {
```

Comment:
Needs a method comment.
What is this trying to measure?

There are lots of kinds of time that are relevant.
- elapsed wall clock time
- total CPU time (e.g. adding up all time spent by all threads, in case we ever multithread the code)
- calendar time

See http://man7.org/linux/man-pages/man7/time.7.html 

Should this really be called an IntervalTimer?